React/Views/RCTTextField.m

```diff
@@ -42,6 +42,25 @@ - (void)setText:(NSString *)text
   }
 }
 
+- (void)_setupPlaceholder {
```

Comment:
It should disappear, but it doesn't really matter if it's set to nil or an empty string, or whether the placeholder label is hidden, since the label would be invisible if it had no text in it. Whatever's easiest to implement.